batcher-bot/liquidity/src/bot.ts

```diff
@@ -0,0 +1,231 @@
+import { HubConnection } from "@microsoft/signalr";
+import {
+  contract_details,
+  liquidity_settings,
+  batch_provision,
+  token_pair,
+} from "./types";
+import { parse_deposit } from "./utils";
+import { submit_deposit, submit_redemption } from "./submitter";
+import { can_provision_always_on, can_provision_jit } from "./provision";
+import { TezosToolkit } from "@taquito/taquito";
+
+const redeem_on_cleared = (
+  batcher_contract: string,
+  msg: any,
+  tezos: TezosToolkit
+) => {
+  for (let i = 0; i < Object.keys(msg.data).length; i++) {
+    try {
+      const message = msg.data[i];
+      if (message.path == "batch_set.batches") {
+        const val = message.content.value;
+        const batch_number = val.batch_number;
+        const status = Object.keys(val.status)[0];
+        if (status == "cleared") {
+          console.info(`Batch ${batch_number} was cleared. Redeeming `);
+          submit_redemption(batcher_contract, tezos).then(() => {
+            console.info(`Redeemed holdings `);
+          });
+        }
+      }
+    } catch (error: any) {
+      console.info("Error parsing bigmap for redemption", error);
+      console.error(error);
+    }
+  }
+};
+const getPairName = (fromName: string, toName: string) => {
+  if (fromName > toName) {
+    return fromName + "/" + toName;
+  }
+
+  return toName + "/" + fromName;
+};
```

Comment:
This can be refactored into:
```ts
const getPairName = (fromName: string, toName: string) => fromName > toName ? fromName + "/" + toName : toName + "/" + fromName;
```

Also I'm not sure to get why you're doing a `>` comparison with 2 strings? If it's really what you want to do, imo it deserves some comments to explain the reason

Reply:
This is a lexicographic comparison no? @JulesGuesnon

Reply:
Actually as it's quite uncommon to see that kind of comparison, it's the first time I hear about it. My bad then, we can close this